capstone/src/main/java/com/google/sps/servlets/Student.java

```diff
@@ -8,10 +8,9 @@
   private int gradYear;
   private String major;
   private String email;
-  private ImmutableList<String> clubs;
+  private ImmutableList<Club> clubs;
```

Comment:
Correct me if I'm wrong, but wasn't the idea behind this being a list of club keys so that the Student object would be as lightweight as possible? And if you wanted to access info about one of the clubs a student is in, you would fetch that info on an as-needed basis? Could you explain the reasoning behind this change a bit more?

Reply:
Yes, you're right about making the Student object as light as possible! However, since we need to access each club's announcements for the student's inbox in the JS fetch method, I wasn't sure whether it would be better to put in the entire Club object as one variable or to have two separate Student object variables (one for the club's name and one for all of the student's clubs' announcements). Would the second approach be better?

Reply:
Is there a way to return both the Student object and the list of announcements separately, but still have them "linked" when fetched in JS? Or some other way to fetch a specific club's announcments in JS? If so, I think that would be the preferred approach.

The problem with this approach is once you have Datastore setup for your Student and Club objects, because they reference each other, if a Club object changes, you'll need to update ALL of the Student objects that have that Club which can introduce a lot of unnecessary operations.

Reply:
I believe this approach is what you meant by separate but linked, but please correct me if I'm wrong: I created a StudentInfo object which stores the Student and announcements separately. Then, when I convert this object to JSON, I can still fetch all the same content as if they were "linked" in the JS file.

Reply:
Yeah, I think this is a lot better. Nice job!

Reply:
Thank you for bringing this up!